core/controllers/editor.py

```diff
@@ -634,53 +632,9 @@ def post(self, entity_type, entity_id):
         filename_prefix = self.payload.get('filename_prefix')
         if filename_prefix is None:
             filename_prefix = self._FILENAME_PREFIX
-        if not raw:
-            raise self.InvalidInputException('No image supplied')
-        if len(raw) > self.HUNDRED_KB_IN_BYTES:
-            raise self.InvalidInputException(
-                'Image exceeds file size limit of 100 KB.')
-        allowed_formats = ', '.join(
-            list(feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS.keys()))
-        if html_validation_service.is_parsable_as_xml(raw):
-            file_format = 'svg'
-            invalid_tags, invalid_attrs = (
-                html_validation_service.get_invalid_svg_tags_and_attrs(raw))
-            if invalid_tags or invalid_attrs:
-                invalid_tags_message = (
-                    'tags: %s' % invalid_tags if invalid_tags else '')
-                invalid_attrs_message = (
-                    'attributes: %s' % invalid_attrs if invalid_attrs else '')
-                raise self.InvalidInputException(
-                    'Unsupported tags/attributes found in the SVG:\n%s\n%s' % (
-                        invalid_tags_message, invalid_attrs_message))
-        else:
-            # Verify that the data is recognized as an image.
-            file_format = imghdr.what(None, h=raw)
-            if file_format not in feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS:
-                raise self.InvalidInputException('Image not recognized')
-
-        # Verify that the file type matches the supplied extension.
-        if not filename:
-            raise self.InvalidInputException('No filename supplied')
-        if filename.rfind('.') == 0:
-            raise self.InvalidInputException('Invalid filename')
-        if '/' in filename or '..' in filename:
-            raise self.InvalidInputException(
-                'Filenames should not include slashes (/) or consecutive '
-                'dot characters.')
-        if '.' not in filename:
-            raise self.InvalidInputException(
-                'Image filename with no extension: it should have '
-                'one of the following extensions: %s.' % allowed_formats)
-
-        dot_index = filename.rfind('.')
-        extension = filename[dot_index + 1:].lower()
-        if (extension not in
-                feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS[file_format]):
-            raise self.InvalidInputException(
-                'Expected a filename ending in .%s, received %s' %
-                (file_format, filename))
 
+        file_format = html_validation_service.validate_image_and_filename(
```

Comment:
I don't think `html_validation_service` is a good place for `validate_image_and_filename` method. Any service which validates files would be a better place (maybe fs_services or a new image service/validation file ?)

Reply:
I couldn't put it in fs_services, since the checks require html_validation_service in it anyway and it caused a circular dependency and since some other image related checks were in there, I put it in html_validation_service.

Reply:
But this, logically, has nothing to do with HTML validation, so you might need a separate (new) service -- image_validation_service.py perhaps.

Reply:
Done, created the new service.